x-pack/test/defend_workflows_cypress/agent.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ToolingLog } from '@kbn/tooling-log';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ChildProcess, spawn } from 'child_process';
+import { getLatestVersion } from './artifact_manager';
+import { Manager } from './resource_manager';
+
+export interface AgentManagerParams {
+  user: string;
+  password: string;
+  kibanaUrl: string;
+  esHost: string;
+  esPort: string;
+}
+
+export class AgentManager extends Manager {
```

Comment:
This seems something generic to me, what about promoting this to an upper level so others can use it?